DCF_Demo.md

```diff
@@ -554,4 +563,605 @@ Receive direct method: cipher.1.decrypt
+<summary>Click here for Powershell commands</summary>
+<br>
+
+The commands are different for Powershell because you have to use ` to escape any " double quotes
+
+<br>
+
+Query for existing interfaces on the device 
+```
+az iot hub invoke-device-method -n [name-of-iothub] -d [name-of-device] --mn "ipc_query.1.query" --mp "{}"
+
+// expected outcome
+{
+  "payload": {
+    "continuation_token": 655615,
+    "result": [
+      "ipc_query.1",
+      "dm.1"
+    ]
+  },
+  "status": 200
+}
+
+```
+
+
+Install key_vault_v1i1 package in the address 134545408 [0x08050000] 
+```
+az iot hub invoke-device-method -n [name-of-iothub] -d [name-of-device] --mn "dm.1.install" --mp "{\`"source_type\`":0,\`"address\`":134545408,\`"package_name\`":\`"key_vault_v1i1\`"}" 
+
+// expected outcome
+{
+  "payload": {},
+  "status": 200
+}
+```
+
+Query for existing interfaces on the device. You should be able to see the newly installed key-vault.
+```
+az iot hub invoke-device-method -n [name-of-iothub] -d [name-of-device] --mn "ipc_query.1.query" --mp "{}"
+
+// expected outcome
+{
+  "payload": {
+    "continuation_token": 655615,
+    "result": [
+      "ipc_query.1",
+      "dm.1",
+      "key_vault.1"
+    ]
+  },
+  "status": 200
+}
+```
+
+</details>
+
+## Creating your own package
+
+The following steps create a new package to be installed in the device. This package will implements a second version of the sprinkler that will expose the interface sprinkler.1.
+
+### Infrastructure
+
+Create a new directory called sprinkler_v2i1 inside of the directory "packages".
+
+```bash
+cd packages
+mkdir sprinkler_v2i1
+cd sprinkler_v2i1
+```
+
+Similar to key_vault, the sprinkler needs a startup code. Copy the entire directory key_vault_v1i1/startup to sprinkler_v2i1.
+
+```bash
+copy -r ..\key_vault_v1i1\startup\* startup
+```
+
+Because DM does not support code with independent-position position yet, you need to change the target address of the package, so, edit the file "startup\STM32L475xx_FLASH.ld" to change the address 0x0805000 to 0x080570000.
+
+```
+MEMORY
+{
+  FLASH (rx) : ORIGIN = 0x08057000, LENGTH = 0x00006FFF
+  RAM   (wx) : ORIGIN = 0, LENGTH = 0
+}
+```
+And change the Flash segment start and end.  
+```
+  __FLASH_segment_start__ = 0x08057000;
+  __FLASH_segment_end__   = 0x0805DFFF;
+```
+
+### Interface
+
+The interface shall be composed by a description of the model and a concrete implementation, both following the DTDL definition. 
+
+The sprinkler_v2i1 will use the same interface of sprinkler_v1i1, the main difference will be in the business logic, the version 2 will be able to control 3 different areas independently, while version 1 can control only 1 area.
+
+#### Model
+
+The model shall describe the interface, including its name and version, and a list of capabilities with the input and output arguments for each capability.
+
+Create a file called sprinkler_1_model.h with the following content in the directory "sprinkler_v2i1".
+
+```c
```

Comment:
Would it make sense to put this in a "package_to_create" directory so-as not to make the md too big?